src/hevm/src/EVM/UnitTest.hs

```diff
@@ -167,20 +167,23 @@ execTest UnitTestOptions { .. } method args = do
 
 checkFailures :: UnitTestOptions -> ABIMethod -> AbiValue -> Bool -> Stepper Bool
 checkFailures UnitTestOptions { .. } method args bailed = do
-     -- Decide whether the test is supposed to fail or succeed
-     let shouldFail = "testFail" `isPrefixOf` method
-
-     -- The test subject should be loaded and initialized already
-     addr <- Stepper.evm $ use (state . contract)
-
-     -- Ask whether any assertions failed
-     Stepper.evm popTrace
-     Stepper.evm $ setupCall testParams addr "failed()" args
-     Stepper.note "Checking whether assertions failed"
-     res <- Stepper.execFullyOrFail >>= Stepper.decode AbiBoolType
-     let AbiBool failed = res
-     -- Return true if the test was successful
-     pure (shouldFail == (bailed || failed))
+  if bailed then
+    pure (False)
+  else do
+    -- Decide whether the test is supposed to fail or succeed
+    let shouldFail = "testFail" `isPrefixOf` method
```

Comment:
```suggestion
   -- Decide whether the test is supposed to fail or succeed
  let shouldFail = "testFail" `isPrefixOf` method
  if bailed then
    pure shouldFail
  else do

```

Reply:
do you agree?

Reply:
I don't think so, we shouldn't be expecting vm failures with `testFailBlah`. We could perhaps invent a new category of `testBailBlah`.

Reply:
but reverts are vm failures, as are INVALID opcode or anything. The following test should pass:
```
function testFail() {
  require(false);
}
```
This is the current semantics (or, well, before the bug recently introduced).

Reply:
Good point. We have

```
ContractNotFound
DecodingError
VMFailed
```

so how about only applying this treatement to `VMFailed` and `DecodingError` and `ContractNotFound` stay as always-fail?

Reply:
`ContractNotFound` can be removed actually, it isn't used and we already have `VMError NoSuchContract`. So all we're actually talking about is `DecodingError`.

Reply:
Ok I can see that your suggestion will do that now anyway, as `DecodingError` is only ever raised from this particular block of code.

Reply:
Yeah. tbh, keeping the decoding inside the stepper script feels a little too much and I could see `Stepper.Failure` being removed completely in favor of just VMError. Then we'd just return false here if the decoding was unsuccessful.

Reply:
but that could also be postponed to keep this focused